python/ctsm/test/test_sys_mesh_modifier.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+
+"""System tests for mesh_mask_modifier"""
+
+import os
+import sys
+import re
+import subprocess
+
+import unittest
+import tempfile
+import shutil
+
+import xarray as xr
+
+from ctsm.path_utils import path_to_ctsm_root
+from ctsm import unit_testing
+from ctsm.modify_mesh_mask.mesh_mask_modifier import mesh_mask_modifier
+
+# Allow test names that pylint doesn't like; otherwise hard to make them
+# readable
+# pylint: disable=invalid-name
+# Allow all the instance attributes that we need
+# pylint: disable=too-many-instance-attributes
+# Allow this due to esmf_cmd
+# pylint: disable=line-too-long
+
+
+class TestSysMeshMaskModifier(unittest.TestCase):
+    """System tests for mesh_mask_modifier"""
+
+    def setUp(self):
+        """
+        Obtain path to the existing:
+        - modify_template.cfg file
+        - /testinputs directory and fsurdat_in, located in /testinputs
+        Make /_tempdir for use by these tests.
+        Obtain path and names for the files being created in /_tempdir:
+        Generate mesh_mask_in.nc applying nco/esmf commands on fsurdat_in.
+        Generate landmask.nc applying nco commands on fsurdat_in.
+        """
+        # Obtain various paths and make /_tempdir
+        self._cfg_template_path = os.path.join(
+            path_to_ctsm_root(), "tools/modify_mesh_mask/modify_template.cfg"
+        )
+        testinputs_path = os.path.join(path_to_ctsm_root(), "python/ctsm/test/testinputs")
+        fsurdat_in = os.path.join(
+            testinputs_path,
+            "surfdata_5x5_amazon_16pfts_Irrig_CMIP6_simyr2000_c171214.nc",
+        )
+        self._tempdir = tempfile.mkdtemp()
+        self._cfg_file_path = os.path.join(self._tempdir, "modify_mesh_mask.cfg")
+        self._mesh_mask_in = os.path.join(self._tempdir, "mesh_mask_in.nc")
+        self._mesh_mask_out = os.path.join(self._tempdir, "mesh_mask_out.nc")
+        self._landmask_file = os.path.join(self._tempdir, "landmask.nc")
+        scrip_file = os.path.join(self._tempdir, "scrip.nc")
+        metadata_file = os.path.join(self._tempdir, "metadata.nc")
+
+        # Generate mesh_mask_in from fsurdat_in
+        ncks_cmd = f"ncks --rgr infer --rgr scrip={scrip_file} {fsurdat_in} {metadata_file}"
+        try:
+            subprocess.check_call(ncks_cmd, shell=True)
+        except subprocess.CalledProcessError as e:
+            err_msg = (
+                f"{e} ERROR using ncks to generate {scrip_file} from "
+                + f"{fsurdat_in}; MOST LIKELY SHOULD INVOKE module load nco"
+            )
+            sys.exit(err_msg)
+        # TODO How to invoke 'module load nco' behind the scenes?
+        # TODO How to make the esmf command generic?
+        # TODO How to write the PET0 file in /tempdir or suppress entirely?
+        esmf_cmd = f"/glade/u/apps/ch/opt/esmf-netcdf/8.0.0/intel/19.0.5/bin/bing/Linux.intel.64.mpiuni.default/ESMF_Scrip2Unstruct {scrip_file} {self._mesh_mask_in} 0"
```

Comment:
@ekluzek another question for when we review this PR:
How to make the esmf command generic?

@billsacks I'm copying you here in case you have insight or in case you have a suggestion of another person to talk to.

Reply:
The only way I can think of is to use the cime configure and make this setup for cime machines, which will include setting up for ESMF. This would be similar to what you've done in mksurfdata_esmf. It's a bit overkill in this case, but I think that might be OK. Possibly as part of that you could add the module load for nco that you talk about above.

Reply:
I'm having an interesting experience where
- in the ctsm5.2.mksurfdata branch I can see /cime/tools/configure (git describe returns cime6.0.15)
- in my modify_meshes branch I do not see configure in the /cime/tools directory though I still see Readme.configure (git describe returns cime6.0.40)

Reply:
oh, and the current PR's work is in the modify_meshes branch...

Reply:
cime directory structure changed, configure is in
cime/CIME/scripts/configure

Reply:
Note to self:
./configure will fail in ctsm5.2.mksurfdata branch when we update cime due to a hardwired path in gen_mksurfdata_build.sh

Reply:
This is now in place and working.